picamera2/devices/imx708/imx708.py

```diff
@@ -0,0 +1,46 @@
+import fcntl
+import os
+
+from picamera2 import Picamera2
+from v4l2 import *
+
+HDR_CTRL_ID = 0x009a0915
+
+
+class IMX708:
+    def __init__(self, camera_num: int):
+        self.device_fd = None
+
+        camera_id = next((c['Id'] for c in Picamera2.global_camera_info() if c['Num'] == camera_num), None)
+        if camera_id is None:
+            raise RuntimeError(f'IMX708: Requested camera number {camera_num} cound not be found')
+
+        for i in range(16):
+            test_dir = f'/sys/class/video4linux/v4l-subdev{i}/device'
+            module_dir = f'{test_dir}/driver/module'
+            id_dir = f'{test_dir}/of_node'
+            if os.path.exists(module_dir) and os.path.islink(module_dir) and 'imx708' in os.readlink(module_dir):
+                if os.path.islink(id_dir) and camera_id in os.readlink(id_dir):
+                    self.device_fd = open(f'/dev/v4l-subdev{i}', 'rb+', buffering=0)
+                    break
+
+        if self.device_fd is None:
+            raise RuntimeError('IMX708: Requested camera device node not found')
+
+    def __del__(self):
+        if self.device_fd:
+            self.device_fd.close()
```

Comment:
Does it matter if the device is only closed by the `__del__` method? I might wonder about having a `close()` method that users should be calling. We could also add context manager support so that folks can use `with` and rely on the closing happening automatically.